### ISSUE TITLE
Rename DagRun queue state to avoid confusion with TaskInstance

### ISSUE THREAD

**Issue by eladkal** [apache/airflow#24045]
### Body

Currently both DagRun and TaskInstance use queue as state (which is not related one to another at all)
![Screen Shot 2022-05-31 at 12 38 44](https://user-images.githubusercontent.com/45845474/171168036-bd6a5995-835b-4f3f-8ca9-084a80f846da.png)
![Screen Shot 2022-05-31 at 12 39 22](https://user-images.githubusercontent.com/45845474/171168042-631e59e1-263a-4160-b454-d88ae7a533dc.png)

I find that this is very confusing! because Queue DagRun actually means that the DagRun didn’t even start while queue task means it passed the scheduler and waiting for worker slot.

My suggestion simply not to use the same string (queue) to describe the different meaning.

(*) We also use Running for both DagRuns/tasks but I find this specific one less confusing as it actually means quite the same (something in worker is being executed).

Slack thread: https://apache-airflow.slack.com/archives/CCPRP7943/p1653990216805099

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

**Comment by ei-grad** [comment]
When the task is waiting for worker it is also "didn't even start" in terms of operator execution.

In my experience it is always tricky to get the logic of scheduled/queued TI for the new people, but there was no problems with the DagRun states naming. To disagree with the slack thread - for the data engineer familiar with airflow internals there is no such confusion, or at least it doesn't seem to be critical enough to break the UX (for the data engineers too).

**Comment by ei-grad** [comment]
What would be the new proposed name for this state? "Pending"? This issue doesn't contain any mentions about the new name, and it is not clear would it hurt the UX much without it.

**Comment by uranusjr** [comment]
Pending sounds reasonable to me.

**Comment by Bowrna** [comment]
Is this issue only about renaming the state in UI?
I have not worked on any UI-related issues before. Could I take a stab at this?

**Comment by uranusjr** [comment]
The name “queued” actually comes from the backend, but I think we could make this a UI-only change to avoid needlessly spending time migrating the database. Or at least we can make the UI change now and modify the backend later.

**Comment by bbovenzi** [comment]
Yeah it shouldn't be hard to change any instance of dagrun queued to pending. We may want a new color too, no?

**Comment by eladkal** [comment]
> Or at least we can make the UI change now and modify the backend later.

Now that Airflow 3 is on the table maybe we should reconsider do both together?
